kernel/src/fs/procfs/pid/task/oom_score_adj.rs

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use core::{fmt::Write, sync::atomic::Ordering};
+
+use crate::{
+    fs::{
+        procfs::template::{FileOps, ProcFileBuilder},
+        utils::{Inode, InodeMode},
+    },
+    prelude::*,
+    process::Process,
+};
+
+/// Represents the inode at `/proc/[pid]/task/[tid]/oom_score_adj` (and also `/proc/[pid]/oom_score_adj`).
+pub struct OomScoreAdjFileOps(Arc<Process>);
+
+impl OomScoreAdjFileOps {
+    pub fn new_inode(process_ref: Arc<Process>, parent: Weak<dyn Inode>) -> Arc<dyn Inode> {
+        // Reference: <https://elixir.bootlin.com/linux/v6.16.5/source/fs/proc/base.c#L3386>
+        ProcFileBuilder::new(Self(process_ref), InodeMode::from_bits_truncate(0o644))
+            .parent(parent)
+            .build()
+            .unwrap()
+    }
+}
+
+impl FileOps for OomScoreAdjFileOps {
+    fn data(&self) -> Result<Vec<u8>> {
+        let oom_score_adj = self.0.oom_score_adj().load(Ordering::Relaxed);
+
+        let mut output = String::new();
+        writeln!(output, "{}", oom_score_adj).unwrap();
+        Ok(output.into_bytes())
+    }
+
+    fn write_at(&self, _offset: usize, reader: &mut VmReader) -> Result<usize> {
+        // TODO: Add `read_cstring_until_end` to the `ReadCString` trait and use that method.
+        let (val, read_bytes) = read_i32_from(reader)?;
+        if !(OOM_SCORE_ADJ_MIN..=OOM_SCORE_ADJ_MAX).contains(&val) {
+            return_errno_with_message!(Errno::EINVAL, "the OOM score adjustment is out of range");
+        }
+
+        self.0.oom_score_adj().store(val as i16, Ordering::Relaxed);
```

Comment:
```
> echo 0 > /proc/self/oom_score_adj
echo: write error: permission denied
```

Since we don't really have a OOM killer, adding TODO here should be okay.
```rust
       // TODO: If the new adjustment value is smaller than the smallest
       // adjustment value that the process has ever reached and the writer
       // does not have the SYS_RESOURCE capability, we should fail with
       // EACCES. See
       // <https://elixir.bootlin.com/linux/v6.16.5/source/fs/proc/base.c#L1152>.
```